src/components/data/viewers/FileViewer.js

```diff
@@ -0,0 +1,271 @@
+ *
+ *
+ */
+
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "i18n";
+import { useRouter } from "next/router";
+
+import ids from "./ids";
+import viewerConstants from "./constants";
+import TextViewer from "./TextViewer";
+import StructuredTextViewer from "./StructuredTextViewer";
+import DocumentViewer from "./DocumentViewer";
+import ImageViewer from "./ImageViewer";
+import VideoViewer from "./VideoViewer";
+
+import { useFileManifest, useReadChunk } from "./queries";
+import constants from "constants";
+import NavigationConstants from "common/NavigationConstants";
+
+import {
+    FETCH_FILE_MANIFEST_QUERY_KEY,
+    READ_CHUNK_QUERY_KEY,
+} from "serviceFacades/filesystem";
+import {
+    mimeTypes,
+    getMimeTypeFromString,
+    getViewerMode,
+} from "components/models/MimeTypes";
+import infoTypes from "components/models/InfoTypes";
+
+import { build } from "@cyverse-de/ui-lib";
+
+import {
+    Button,
+    CircularProgress,
+    Toolbar,
+    Typography,
+} from "@material-ui/core";
+
+const VIEWER_TYPE = {
+    PLAIN: "plain",
+    STRUCTURED: "structured",
+    IMAGE: "image",
+    VIDEO: "video",
+    DOCUMENT: "document", //pdf,xml etc...
+};
+
+export default function FileViewer(props) {
+    const { path, resourceId, baseId } = props;
+
+    const { t } = useTranslation("data");
+    const router = useRouter();
+    const [mode, setMode] = useState(null);
+    const [readChunkKey, setReadChunkKey] = useState(READ_CHUNK_QUERY_KEY);
+    const [readChunkQueryEnabled, setReadChunkQueryEnabled] = useState(false);
+    const [viewerType, setViewerType] = useState(VIEWER_TYPE.PLAIN);
+
+    const {
+        isFetching,
+        data: manifest,
+        error: manifestError,
+    } = useFileManifest(
+        [FETCH_FILE_MANIFEST_QUERY_KEY, path],
+        path !== null && path !== undefined
+    );
+
+    const {
+        status,
+        data,
+        isFetchingMore,
+        fetchMore,
+        canFetchMore,
+        error: chunkError,
+    } = useReadChunk(
+        readChunkKey,
+        readChunkQueryEnabled,
+        (lastGroup, allGroups) => {
+            const totalPages = Math.ceil(
+                lastGroup["file-size"] / viewerConstants.DEFAULT_PAGE_SIZE
+            );
+            if (allGroups.length < totalPages) {
+                return allGroups.length;
+            } else {
+                return false;
+            }
+        }
+    );
+    const getColumnDelimiter = (infoType) => {
+        if (infoTypes.CSV === infoType) {
+            return viewerConstants.COMMA_DELIMITER;
+        } else if (
+            infoTypes.TSV === infoType ||
+            infoTypes.VCF === infoType ||
+            infoTypes.GFF === infoType ||
+            infoTypes.BED === infoType ||
+            infoTypes.GTF === infoType ||
+            infoTypes.BOWTIE === infoType ||
+            infoTypes.HT_ANALYSIS_PATH_LIST === infoType ||
+            infoTypes.MULTI_INPUT_PATH_LIST === infoType
```

Comment:
It might be nice to change this to use [Array.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) if we can. Babel will probably swap it out for he sake of the few browsers that don't support it, but it makes this code a lot smaller/easier to read, I suspect.